src/app/shared/components/users-list/users-list.component.ts

```diff
@@ -85,15 +87,15 @@ export class UsersListComponent implements OnInit, AfterViewInit, OnChanges {
     }
   }
 
-  onDelete(user: ProviderAdminTable): void {
-    this.deleteAdmin.emit(user);
+  onDelete(user: UsersTable): void {
+    this.delete.emit(user);
   }
 
-  onBlock(user: ProviderAdminTable): void {
-    this.blockAdmin.emit(user);
+  onBlock(user: BlockData): void {
+   user.isBlocked ? this.unblock.emit(user) : this.block.emit(user);
```

Comment:
if you check here isBlocked property, then why do you pass an additional value from the template?

Reply:
i removed this check and removed value output unblock. For now, we have only one value onblock and we do one check in method openBlockModal